$OpenBSD: patch-lib_src_facts_openbsd_networking_resolver_cc,v 1.1 2016/05/30 15:07:58 jasper Exp $

- execution.hpp moved to leatherman
- Implement missing get_link_address_bytes()

--- lib/src/facts/openbsd/networking_resolver.cc.orig	Fri May 20 18:36:55 2016
+++ lib/src/facts/openbsd/networking_resolver.cc	Mon May 30 16:57:25 2016
@@ -1,6 +1,6 @@
 #include <internal/facts/openbsd/networking_resolver.hpp>
 #include <internal/util/bsd/scoped_ifaddrs.hpp>
-#include <facter/execution/execution.hpp>
+#include <leatherman/execution/execution.hpp>
 #include <leatherman/logging/logging.hpp>
 #include <boost/algorithm/string.hpp>
 #include <sys/sockio.h>
@@ -12,13 +12,25 @@
 using namespace std;
 using namespace facter::util;
 using namespace facter::util::bsd;
-using namespace facter::execution;
+using namespace leatherman::execution;
 
 namespace facter { namespace facts { namespace openbsd {
 
     bool networking_resolver::is_link_address(sockaddr const* addr) const
     {
         return addr && addr->sa_family == AF_LINK;
+    }
+
+    uint8_t const* networking_resolver::get_link_address_bytes(sockaddr const* addr) const
+    {
+        if (!is_link_address(addr)) {
+            return nullptr;
+        }
+        sockaddr_dl const* link_addr = reinterpret_cast<sockaddr_dl const*>(addr);
+        if (link_addr->sdl_alen != 6) {
+            return nullptr;
+        }
+        return reinterpret_cast<uint8_t const*>(LLADDR(link_addr));
     }
 
     boost::optional<uint64_t> networking_resolver::get_link_mtu(string const& interface, void* data) const
